lively.ide/studio/sidebar-flap.js

```diff
@@ -0,0 +1,78 @@
+import { disconnect, connect } from 'lively.bindings';
+import { defaultPropertiesPanelWidth } from './properties-panel.cp.js';
+import { easings } from 'lively.morphic';
+import { pt } from 'lively.graphics';
+export async function toggleSidebar () {
+  const world = this.world();
+  const sidebarIsFadingOut = world.get(this.target);
+  const targetSidebar = await world.openSideBar(this.target);
+  if (sidebarIsFadingOut) {
+    this.view.animate({
+      left: (this.target === 'scene graph') ? 0 : world.visibleBounds().width - this.view.width,
+      duration: 300
+    });
+
+    if (this.target === 'scene graph') {
+      disconnect(targetSidebar, 'extent', this.view, 'left');
+    }
+  } else {
+    const sideBarWidth = targetSidebar?.width || defaultPropertiesPanelWidth;
+    const dragAreaWidth = 5;
+    const left = (this.target === 'scene graph')
+      ? sideBarWidth - dragAreaWidth
+      : world.visibleBounds().width - sideBarWidth - this.view.width;
+    if (this.target === 'scene graph') {
+      connect(targetSidebar, 'extent', this.view, 'left', {
+        updater: ($upd, extent) => {
+          $upd(extent.x - dragAreaWidth);
+        },
+        varMapping: { dragAreaWidth }
+      });
+    }
+    this.view.animate({
+      left,
+      duration: 300,
+      easing: easings.outCirc
+    });
+  }
+}
+
+export function relayoutSidebarFlapInWorld () {
+  if (this.target === 'scene graph') return;
+  const propertiesPanelExists = $world.get(this.target);
+  if (propertiesPanelExists) this.view.position = this.view.position.withX($world.visibleBounds().width - defaultPropertiesPanelWidth - this.view.width);
+  else this.view.position = this.view.position.withX($world.visibleBounds().width - this.view.width);
+}
+
+export async function openSidebarFlapInWorld () {
+  const world = this.world();
+  const { view } = this;
```

Comment:
I gotta say, I would prefer this to be passed as an argument. I know this is technically possible but it just is super confusing to me to assume that when we are executing we are bound against a flap model.